kernel/src/vm/vmar/mod.rs

```diff
@@ -690,6 +705,114 @@ impl Vmar_ {
         &self.vm_space
     }
 
+    fn read_maybe_remote(&self, vaddr: Vaddr, writer: &mut VmWriter) -> Result<usize> {
+        let len = writer.avail();
+        let read = |frame: UFrame, skip_offset: usize| {
+            let mut reader = frame.reader();
+            reader.skip(skip_offset);
+            let read_bytes = reader.read_fallible(writer)?;
```

Comment:
Page faults here will cause dead locks if we're accessing the current VMAR using `read_remote`. We can't lock the same VMAR twice.